app/actions/websocket/channel.ts

```diff
@@ -1,22 +1,86 @@
+import {prepareCommonSystemValues, queryConfig, setCurrentChannelId} from '@queries/servers/system';
 import {queryLastChannelFromTeam} from '@queries/servers/team';
-import {queryCurrentUser} from '@queries/servers/user';
+import {queryCurrentUser, queryUserById} from '@queries/servers/user';
 import {dismissAllModals, popToRoot} from '@screens/navigation';
 import {isTablet} from '@utils/helpers';
-import {isGuest} from '@utils/user';
 
-export async function handleUserRemovedEvent(serverUrl: string, msg: any) {
+export async function handleUserAddedToChannelEvent(serverUrl: string, msg: any) {
+    const database = DatabaseManager.serverDatabases[serverUrl];
+    if (!database) {
+        return;
+    }
+    const currentUser = await queryCurrentUser(database.database);
+    const {team_id: teamId, channel_id: channelId, user_id: userId} = msg.data;
+
+    const models: Model[] = [];
+
+    try {
+        const addedUser = queryUserById(database.database, userId);
+        if (!addedUser) {
+            // TODO Potential improvement https://mattermost.atlassian.net/browse/MM-40581
+            const {users} = await fetchUsersByIds(serverUrl, [userId], true);
+            if (users) {
+                models.push(...await database.operator.handleUsers({users, prepareRecordsOnly: true}));
+            }
+        }
+
+        if (userId === currentUser?.id) {
+            const {channels, memberships} = await fetchMyChannel(serverUrl, teamId, channelId, true);
+            if (channels && memberships) {
+                const prepare = await prepareMyChannelsForTeam(database.operator, teamId, channels, memberships);
+                if (prepare) {
+                    const prepareModels = await Promise.all(prepare);
+                    const flattenedModels = prepareModels.flat();
+                    if (flattenedModels?.length > 0) {
+                        models.push(...flattenedModels);
+                    }
+                }
+            }
+
+            const {posts, order, authors, actionType, previousPostId} = await fetchPostsForChannel(serverUrl, channelId, true);
+            if (posts?.length && order && actionType) {
+                models.push(...await database.operator.handlePosts({
+                    actionType,
+                    order,
+                    posts,
+                    previousPostId,
+                    prepareRecordsOnly: true,
+                }));
+            }
+
+            if (authors?.length) {
+                models.push(...await database.operator.handleUsers({users: authors, prepareRecordsOnly: true}));
+            }
+
+            database.operator.batchRecords(models);
+        } else {
+            const channels = await queryChannelsById(database.database, [channelId]);
+            if (channels?.[0]) {
+                models.push(...await database.operator.handleChannelMembership({channelMemberships: [{channel_id: channelId, user_id: userId}], prepareRecordsOnly: true}));
```

Comment:
Similar to the other PR with regards to the long chains ... and is it safe to push the await like that ?
``` 
                models.push(...await database.operator.handleChannelMembership({channelMemberships: [{channel_id: channelId, user_id: userId}], prepareRecordsOnly: true}));
```

Reply:
This should be safe. The handleX functions will always return a list of models, even if empty. And `models.push(...[])` keeps the original list the same.

I will add a few line breaks to keep the lines shorter.